AsyncDisplayKit/ASCellNode.h

```diff
@@ -30,6 +30,14 @@ typedef NS_ENUM(NSUInteger, ASCellNodeVisibilityEvent) {
    * Indicates a cell is no longer visible
    */
   ASCellNodeVisibilityEventInvisible,
+  /**
+   * Indicates user has started dragging the visible cell
+   */
+  ASCellNodeVisibilityEventWillBeginDragging,
```

Comment:
My main concern with this is that these events will be triggered even if the user is not dragging on the cell itself, which is not particularly clear from the API.  Also, if a cell becomes invisible in the process of the user driving on the screen before their fingerless the screen, will we send a did and dragging to that cell?

That would be very important, as well as the reverse, where a cell that is not yet visible gets a begin call even if the Touch is already on screen and continuing to drag at the moment it becomes visible. Making these calls imbalance could create issues for a variety of clients.